### DISCUSSION TITLE
How to Enable 3rd Party Python Markdown Extensions

### DISCUSSION THREAD

**Discussion by RoyiAvital** [facelessuser/MarkdownPreview#131] [Q&A] [answered]
Is there a simple guide how to enable 3rd party (Not `pymdown-extensions`) inside `Markdown Preview`?

As a test case I suggest https://github.com/Evidlo/markdown_captions.
Also, it would be nice to define parameters. In the case above there aren't but let's assume there is `param1` we want to set to `true.

**Remark**: Related to https://github.com/facelessuser/pymdown-extensions/issues/1341.

**Comment by facelessuser** [comment, marked as answer]
I'll try and take a look at this sometime soon. Many extensions may work fine, some may require some backporting (Sublime is on Python 3.3). Though, at some point in the future, I'm going to update Markdown Preview to use ST4's new Python 3.8 framework.

Whether we use `markdown_captions` as an example or some toy example, we can most likely put together a real-world, testable example.

**Reply by facelessuser** [reply]
While I don't have documentation up. If you wanted to use Markdown Captions extension, simply download the zip, copy the `markdown_captions` folder that contains the source code. I just downloaded the master branch and copied the folder `markdown_captions-master/markdown_captions` to Sublime's `Packages` folder.

And then added the plugin the to the settings:

```js
"markdown_extensions": [
        "markdown.extensions.footnotes",
        "markdown.extensions.attr_list",
        "markdown.extensions.def_list",
        "markdown.extensions.tables",
        "markdown.extensions.abbr",
        "pymdownx.highlight",
        "pymdownx.betterem",
        "markdown_captions.markdown_captions", // <-- Added here
        // more extensions
```

That's it. Worked fine.

**Reply by RoyiAvital** [reply]
That's great!

So in case of parameters just use the same from as with other packages?

**Reply by facelessuser** [reply]
Yeah, it's all generalized. There's nothing special about the already shipped plugins vs. user installed.

**Reply by RoyiAvital** [reply]
Created a Wiki as a reference for others: https://github.com/facelessuser/MarkdownPreview/wiki/Using-3rd-Party-Python-Markdown-Extensions.
